prime_divisors/primes.rb

```diff
@@ -1,3 +1,17 @@
+# This program prints the value of "pi(n)", i.e. the numbers of primes
+# less than @p n.
+#
+# Arguments:
+#
+#     n: The number less than equal to which primes must be computed.
+#
+# Result:
+#
+#     n pi(n)
+#
+
+# @return The number of prime divisors of @p n.
+
 def n_prime_divisors(n)
```

Comment:
What bs. The function returns 4 given 4.
On Jan 20, 2015 12:36 PM, "Manav" <notifications@github.com> wrote:

> In prime_divisors/primes.rb
> <https://github.com/zbard/benchmark/pull/3#discussion-diff-23205579>:
>
> > @@ -1,3 +1,17 @@
> > +# This program prints the value of "pi(n)", i.e. the numbers of primes
> > +# less than @p n.
> > +#
> > +# Arguments:
> > +#
> > +#     n: The number less than equal to which primes must be computed.
> > +#
> > +# Result:
> > +#
> > +#     n pi(n)
> > +#
> > +
> > +# @return The number of prime divisors of @p n.
> > +
> >  def n_prime_divisors(n)
>
>  For the sake of posterity, this doesn't calculate len(prime_divisors).
> It fails to account for numbers having a divisor greater than sqrt(n). It
> does return 2 for primes, which is what you want - although to be be
> pedantic, 1 is not a prime.
>
> Too much math for me.
>
> You could've also change the ruby program then to match the original
> semantics if you wanted.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zbard/benchmark/pull/3/files#r23205579>.
>